### ISSUE TITLE
🟢 Optimize Lambda Functions (Memory, Timeout, Cost)

### ISSUE THREAD

**Issue by prakashdamo** [prakashdamo/mrktdly#14]
**Priority:** 🟢 Nice to Have - Backlog
**Labels:** optimization, infrastructure, cost-savings
**Estimate:** 30 minutes
**Potential Savings:** $5-10/month

## Current State
All Lambda functions are well-architected with small package sizes (<2MB). However, some may be over-provisioned.

### Memory Usage
| Function | Current | Recommendation |
|----------|---------|----------------|
| market-insights | 1024MB | Test with 512MB |
| features | 512MB | Monitor, likely correct |
| swing-scanner | 512MB | Monitor, likely correct |
| Others | 128MB | ✅ Optimal |

### Timeout Settings
| Function | Current | Actual Usage | Recommendation |
|----------|---------|--------------|----------------|
| features | 300s | Unknown | Monitor, reduce if <120s |
| swing-scanner | 300s | Unknown | Monitor, reduce if <120s |
| Others | 30-120s | ✅ Reasonable | Keep as-is |

### 1. Monitor Actual Usage (Week 1)
- [ ] Check CloudWatch logs for actual duration
```bash
aws logs filter-log-events --log-group-name /aws/lambda/mrktdly-features \
  --filter-pattern "REPORT" --region us-east-1 | grep Duration
```
- [ ] Check memory usage patterns
- [ ] Document findings in spreadsheet

### 2. Reduce Timeouts (If Applicable)
- [ ] If features averages <60s, reduce timeout to 120s
- [ ] If swing-scanner averages <60s, reduce timeout to 120s
```bash
aws lambda update-function-configuration \
  --function-name mrktdly-features \
  --timeout 120 --region us-east-1
```

### 3. Test Lower Memory
- [ ] Reduce market-insights from 1024MB to 512MB
- [ ] Monitor for errors or timeouts
- [ ] Rollback if issues occur
```bash
aws lambda update-function-configuration \
  --function-name mrktdly-market-insights \
  --memory-size 512 --region us-east-1
```

### 4. Clean Up Local Directories
- [ ] Remove sklearn-layer.zip files (not needed locally)
- [ ] Remove training data from features directory
- [ ] Remove .pkl files (training artifacts)
```bash
rm /home/prakash/marketdly/lambda/ml-predictions/sklearn-layer.zip
rm /home/prakash/marketdly/lambda/state-classifier/sklearn-layer.zip
rm /home/prakash/marketdly/lambda/features/training_data.csv
rm /home/prakash/marketdly/lambda/features/*.pkl
```

## Current Costs (Estimated)
- Lambda invocations: $5-10/month
- Lambda compute: $10-20/month
- S3 model storage: $6/month
- **Total: ~$20-35/month**

## Expected Savings
- Reduce memory: $3-5/month
- Reduce timeouts: $2-3/month
- **Total savings: $5-10/month**

## What's Already Good ✅
- Small package sizes (<2MB)
- Models in S3 (not in packages)
- Most functions use 128MB (optimal)
- Appropriate use of layers

## When to Do This
- **Not urgent** - System is already efficient
- **After launch** - Focus on growth first
- **During optimization phase** - Week 15-18 in roadmap

## Success Metrics
- [ ] Reduced monthly Lambda costs by $5-10
- [ ] No increase in errors or timeouts
- [ ] Documented actual usage patterns

## Notes
This is a cost optimization, not a performance issue. Current architecture is solid.
